types/react-transition-group/Transition.d.ts

```diff
@@ -0,0 +1,24 @@
+import { Component, HTMLProps } from "react";
+import { TransitionActions } from "react-transition-group";
+
+export type EndHandler = (node: HTMLElement, done: () => void) => void;
+export type EnterHandler = (node: HTMLElement, isAppearing: boolean) => void;
+export type ExitHandler = (node: HTMLElement) => void;
+
+export interface TransitionProps extends TransitionActions, HTMLProps<any> {
```

Comment:
`children` has to be optional even though it is required. You can specify children elements both as direct JSX elements nested under wrapping element or as a regular `children` property on the JSX element. You can't validate both at the same time.